converted to array-based todo list

mmarganski opened the pull request.

mmarganski committed:
converted to array-based todo list

jpudysz commented on myscripts.js:
```diff
@@ -0,0 +1,173 @@
+const toDoObjectsArray =[]
```
enter after const, try to separate assignments from instructions

jpudysz replied:
space after `=`

jpudysz commented on myscripts.js:
```diff
@@ -0,0 +1,173 @@
+const toDoObjectsArray =[]
+document.getElementById('mainSubmitButton').setAttribute('onclick','newToDoItem()')
```
no need for a string inside `setAttribute`. I more than sure that you can use `setAttribute('onclick', newToDoItem)`

jpudysz commented on myscripts.js:
```diff
@@ -0,0 +1,173 @@
+const toDoObjectsArray =[]
+document.getElementById('mainSubmitButton').setAttribute('onclick','newToDoItem()')
+document.getElementById('clearAllButton').setAttribute('onclick','clearAll()')
```
aa = as above

jpudysz commented on myscripts.js:
```diff
@@ -0,0 +1,173 @@
+
+class toDoObject{
+    constructor(input) {
+        this.id = String(Math.random().toFixed(6) * 1000000)
```
you can create private function (separated) to create ID

jpudysz commented on myscripts.js:
```diff
@@ -0,0 +1,173 @@
+        this.id = String(Math.random().toFixed(6) * 1000000)
+        this.isDone = false
+        this.text = input
+    }
```
enter after each method

jpudysz commented on myscripts.js:
```diff
@@ -0,0 +1,173 @@
+        this.text = input
+    }
+    check() {
+        this.isDone === true
```
its professionally called `toggle()`

jpudysz commented on myscripts.js:
```diff
@@ -0,0 +1,173 @@
+    }
+    check() {
+        this.isDone === true
+            ? this.isDone = false
```
additionally no need to cast it to Boolean you can use:
```
this.isDone = !this.isDone
```

jpudysz commented on myscripts.js:
```diff
@@ -0,0 +1,173 @@
+            ? this.isDone = false
+            : this.isDone = true
+    }
+    changeText(input) {
```
aa (spacing)

jpudysz commented on myscripts.js:
```diff
@@ -0,0 +1,173 @@
+
+const clearAll = () => {
+    if (confirm("Are you sure you want to clear every task?")) {
+        const allTasks = document.querySelectorAll("#toDoList li")
```
enter after const

jpudysz commented on myscripts.js:
```diff
@@ -0,0 +1,173 @@
+const clearAll = () => {
+    if (confirm("Are you sure you want to clear every task?")) {
+        const allTasks = document.querySelectorAll("#toDoList li")
+        toDoObjectsArray.splice(0,toDoObjectsArray.length)
```
`toDoObjectArray = []`

jpudysz commented on myscripts.js:
```diff
@@ -0,0 +1,173 @@
+    if (confirm("Are you sure you want to clear every task?")) {
+        const allTasks = document.querySelectorAll("#toDoList li")
+        toDoObjectsArray.splice(0,toDoObjectsArray.length)
+        allTasks.forEach((task) => deleteItem(task))
```
`allTasks.forEach(deleteItem)`

jpudysz commented on myscripts.js:
```diff
@@ -0,0 +1,173 @@
+}
+
+const newToDoItem = () => {
+    const input = document.getElementById('mainInput').value
```
enter

jpudysz replied:
`const inputValue` <- you must be more descriptive

jpudysz commented on myscripts.js:
```diff
@@ -0,0 +1,173 @@
+    }
+}
+
+const clearAll = () => {
```
be more descriptive `clearAllTasks`

jpudysz commented on myscripts.js:
```diff
@@ -0,0 +1,173 @@
+    }
+}
+
+const newToDoItem = () => {
```
createNewTodo

jpudysz commented on myscripts.js:
```diff
@@ -0,0 +1,173 @@
+
+const newToDoItem = () => {
+    const input = document.getElementById('mainInput').value
+    if(input.length>0) {
```
spacing between `>`

jpudysz commented on myscripts.js:
```diff
@@ -0,0 +1,173 @@
+const newToDoItem = () => {
+    const input = document.getElementById('mainInput').value
+    if(input.length>0) {
+        const newToDoObject = new toDoObject(input)
```
I would prefer clean objects as class is a next step. But you can leave it

jpudysz commented on myscripts.js:
```diff
@@ -0,0 +1,173 @@
+        const newToDoObject = new toDoObject(input)
+
+        document.getElementById("mainInput").value = ""
+        toDoObjectsArray.push([newToDoObject.id,newToDoObject])
```
`push` is mutating array, I would prefer spread

jpudysz commented on myscripts.js:
```diff
@@ -0,0 +1,173 @@
+    }
+}
+
+const editItem = currentListElement =>{
```
space after `=>`
also you can rename it to `editItem`

jpudysz commented on myscripts.js:
```diff
@@ -0,0 +1,173 @@
+    showElement('saveEditButton',currentListElement)
+}
+
+const saveEdit = currentListElement => {
```
`const editTodoText`

jpudysz commented on myscripts.js:
```diff
@@ -0,0 +1,173 @@
+    showElement('checkBox',currentListElement)
+    showElement('textField',currentListElement)
+
+    const newText = currentListElement.getElementsByClassName('inputField')[0].value
```
try to use destructuring

jpudysz commented on myscripts.js:
```diff
@@ -0,0 +1,173 @@
+    const newText = currentListElement.getElementsByClassName('inputField')[0].value
+    const currentObject = getObjectFromDOMElement(currentListElement)
+
+    currentListElement.querySelector('span').innerText=newText
```
spacing ` = `

jpudysz commented on myscripts.js:
```diff
@@ -0,0 +1,173 @@
+    printItems()
+}
+
+const getObjectFromDOMElement = (currentListElement)=>{
```
no need for `()` if arrow function takes one arg

jpudysz commented on myscripts.js:
```diff
@@ -0,0 +1,173 @@
+
+const getObjectFromDOMElement = (currentListElement)=>{
+    const id = currentListElement.getAttribute('id')
+    return toDoObjectsArray.find(([elementId,element]) => elementId === id)[1]
```
use array destructuring

jpudysz replied:
also this code can crash, you should always use `if` with `find`

jpudysz commented on myscripts.js:
```diff
@@ -0,0 +1,173 @@
+    return toDoObjectsArray.find(([elementId,element]) => elementId === id)[1]
+}
+
+const deleteItem= currentListElement => {
```
space before `=`

jpudysz replied:
`deleteTodo`

jpudysz commented on myscripts.js:
```diff
@@ -0,0 +1,173 @@
+    const currentIndex = toDoObjectsArray.findIndex(([elementId,element]) => elementId === id)
+    const newArray = toDoObjectsArray.filter((element,index) => index !== currentIndex)
+
+    toDoObjectsArray.splice(0,toDoObjectsArray.length)
```
I don't like it, use `filter` and you will be able to remove all this lines

jpudysz commented on myscripts.js:
```diff
@@ -0,0 +1,173 @@
+    toDoObjectsArray.splice(0,toDoObjectsArray.length)
+    newArray.forEach(element => toDoObjectsArray.push(element))
+    currentListElement.remove()
+    if(toDoObjectsArray.length===0){
```
space between ` === `

jpudysz replied:
enter before `if`

jpudysz commented on myscripts.js:
```diff
@@ -0,0 +1,173 @@
+}
+
+
+const check = currentListElement => {
```
`toggleTodo`

jpudysz commented on myscripts.js:
```diff
@@ -0,0 +1,173 @@
+        currentObject.check()
+        printItems()
+    }
+    else {
```
I dont like else, use if + return

jpudysz commented on myscripts.js:
```diff
@@ -0,0 +1,173 @@
+const check = currentListElement => {
+    const currentObject = getObjectFromDOMElement(currentListElement)
+
+    if(currentListElement.getElementsByClassName('checkBox')[0].checked) {
```
use destructuring

jpudysz commented on myscripts.js:
```diff
@@ -0,0 +1,173 @@
+    }
+}
+
+const printItems = () => document.getElementById('debugger').innerText=JSON.stringify([...toDoObjectsArray.entries()])
```
spaces between ` = `

jpudysz commented on myscripts.js:
```diff
@@ -0,0 +1,173 @@
+
+const printItems = () => document.getElementById('debugger').innerText=JSON.stringify([...toDoObjectsArray.entries()])
+
+const hideElement = (name, div) => div.getElementsByClassName(name)[0].setAttribute('hidden','true')
```
destructing

jpudysz commented on myscripts.js:
```diff
@@ -0,0 +1,173 @@
+
+const hideElement = (name, div) => div.getElementsByClassName(name)[0].setAttribute('hidden','true')
+
+const showElement = (name, div) => div.getElementsByClassName(name)[0].removeAttribute('hidden')
```
aa

jpudysz commented on myscripts.js:
```diff
@@ -0,0 +1,173 @@
+
+const showElement = (name, div) => div.getElementsByClassName(name)[0].removeAttribute('hidden')
+
+const hideButtons = currentListElement => currentListElement.querySelectorAll('button').forEach((button) => button.setAttribute('hidden','true'))
```
you can format it better:
```
const hideButtons = currentListElement => currentListElement
         .querySelectorAll('button')
         .forEach((button) => button.setAttribute('hidden','true'))
```
also remove `()` inside `forEeach`

jpudysz commented on myscripts.js:
```diff
@@ -0,0 +1,173 @@
+const showOptions = div => div.removeAttribute('hidden')
+
+const newListElement = (text,id) => {
+
```
remove enter

jpudysz commented on myscripts.js:
```diff
@@ -0,0 +1,173 @@
+    const inputField = document.createElement('input')
+    inputField.setAttribute('hidden','true')
+    inputField.setAttribute('class','inputField')
+    inputField.value=text
```
space between ` = `

jpudysz commented on myscripts.js:
```diff
@@ -0,0 +1,173 @@
+    checkbox.addEventListener('change', () => check(li))
+
+    const inputField = document.createElement('input')
+    inputField.setAttribute('hidden','true')
```
can be simple true, wuthout string

jpudysz commented on myscripts.js:
```diff
@@ -0,0 +1,173 @@
+    const editButton = document.createElement('button')
+    editButton.setAttribute('class','editButton')
+    editButton.addEventListener('click', () => editItem(li))
+    editButton.textContent ='edit'
```
aa

jpudysz commented on myscripts.js:
```diff
@@ -0,0 +1,173 @@
+    saveEditButton.setAttribute('hidden','true')
+    saveEditButton.setAttribute('class','saveEditButton')
+    saveEditButton.addEventListener('click', () => saveEdit(li))
+    saveEditButton.textContent ='save'
```
aa

jpudysz commented on myscripts.js:
```diff
@@ -0,0 +1,173 @@
+    const deleteButton = document.createElement('button')
+    deleteButton.setAttribute('class','deleteButton')
+    deleteButton.addEventListener('click', () => deleteItem(li))
+    deleteButton.textContent='delete'
```
aa

jpudysz commented on myscripts.js:
```diff
@@ -0,0 +1,173 @@
+
+    const textField = document.createElement('span')
+    textField.setAttribute('class','textField')
+    textField.innerText=text
```
aa

jpudysz commented on myscripts.js:
```diff
@@ -0,0 +1,173 @@
+    div.appendChild(saveEditButton)
+    div.appendChild(editButton)
+    div.appendChild(deleteButton)
+    div.setAttribute('hidden','true')
```
can be bool

jpudysz commented on myscripts.js:
```diff
@@ -0,0 +1,173 @@
+    li.appendChild(textField)
+    li.addEventListener('mouseover',() => showOptions(div))
+    li.addEventListener('mouseout',() => {
+        if(inputField.hidden === true) {
```
`if(inputField.hidden)`

jpudysz commented on myscripts.js:
```diff
@@ -0,0 +1,173 @@
+
+
+
+
```
leave one enter

jpudysz commented:
Overall not that bad, try to keep spacing between elements and be more descriptive

mmarganski committed:
refactored;
setAttribute requires DOMString, setAttribute('onclick', newToDoItem) doesn't work; kept setAttribute('onclick','createNewTodo()')

jpudysz commented on index.js:
```diff
@@ -1,115 +1,143 @@
-document.getElementById('clearAllButton').setAttribute('onclick','clearAll()')
+let toDoObjectsArray = []
+
+document.getElementById('mainSubmitButton').setAttribute('onclick','createNewTodo()')
```
still, you can use it without string

jpudysz commented on index.js:
```diff
@@ -1,115 +1,143 @@
+let toDoObjectsArray = []
+
+document.getElementById('mainSubmitButton').setAttribute('onclick','createNewTodo()')
+document.getElementById('clearAllButton').setAttribute('onclick','clearAllTasks()')
```
aa

jpudysz commented on index.js:
```diff
@@ -1,115 +1,143 @@
-            ? this.isDone = false
-            : this.isDone = true
+
+    #createID(){
```
space after `()`

jpudysz commented on index.js:
```diff
@@ -1,115 +1,143 @@
 
-const clearAll = () => {
+const clearAllTasks = () => {
+
```
remove enter

jpudysz commented on index.js:
```diff
@@ -1,115 +1,143 @@
+const createNewTodo = () => {
+    const inputValue = document.getElementById('mainInput').value
+
+    if(inputValue.length) {
```
inputValue.length > 0 was fine as you need to cast number to boolean

jpudysz commented on index.js:
```diff
@@ -1,115 +1,143 @@
+document.getElementById('mainSubmitButton').setAttribute('onclick','createNewTodo()')
+document.getElementById('clearAllButton').setAttribute('onclick','clearAllTasks()')
 
 class toDoObject{
```
class name should be uppercased like ToDo

jpudysz replied:
still the case

jpudysz commented on index.js:
```diff
@@ -1,115 +1,143 @@
 
-    const newText = currentListElement.getElementsByClassName('inputField')[0].value
     const currentObject = getObjectFromDOMElement(currentListElement)
+    const [inputFieldText,] = currentListElement.getElementsByClassName('inputField')
```
no need for `,`

jpudysz commented on index.js:
```diff
@@ -1,115 +1,143 @@
-    currentListElement.querySelector('span').innerText=newText
-    currentObject.changeText(newText)
+    currentListElement.querySelector('span').innerText = inputFieldText.value
+    console.log(inputFieldText.value)
```
remove `console.log`

jpudysz commented on index.js:
```diff
@@ -1,115 +1,143 @@
+const getObjectFromDOMElement = currentListElement => {
     const id = currentListElement.getAttribute('id')
-    return toDoObjectsArray.find(([elementId,element]) => elementId === id)[1]
+    const toDoObjectElement = toDoObjectsArray.find(([elementId,]) => elementId === id)
```
no need for `,`

jpudysz commented on index.js:
```diff
@@ -1,115 +1,143 @@
+        const [,toDoObject] = toDoObjectElement
+
+        return toDoObject
+    }
```
and if not you should return null

jpudysz commented on index.js:
```diff
@@ -1,115 +1,143 @@
 
-    toDoObjectsArray.splice(0,toDoObjectsArray.length)
-    newArray.forEach(element => toDoObjectsArray.push(element))
+    toDoObjectsArray = toDoObjectsArray.filter( arrayElement => {
```
remove extra space before `arrayElement`

jpudysz commented on index.js:
```diff
@@ -1,115 +1,143 @@
+    toDoObjectsArray = toDoObjectsArray.filter( arrayElement => {
+        const [elementId,] = arrayElement
+        return elementId !== id
+    })
```
you can use shortcut:
```
toDoObjectsArray.filter(([elementId]) => elementId !== id)
```

jpudysz commented on index.js:
```diff
@@ -1,115 +1,143 @@
-    if(toDoObjectsArray.length===0){
-        document.getElementById("clearAllButton").setAttribute('hidden','true')
+
+    if(!toDoObjectsArray.length) {
```
`> 0`

jpudysz commented on index.js:
```diff
@@ -1,115 +1,143 @@
-        document.getElementById("clearAllButton").setAttribute('hidden','true')
+
+    if(!toDoObjectsArray.length) {
+        document.getElementById("clearAllButton").setAttribute('hidden',true)
```
space before `true`, give it more spacing

jpudysz commented on index.js:
```diff
@@ -1,115 +1,143 @@
+
+    if(!toDoObjectsArray.length) {
+        document.getElementById("clearAllButton").setAttribute('hidden',true)
     }
```
enter after `if`

jpudysz commented on index.js:
```diff
@@ -1,115 +1,143 @@
-const check = currentListElement => {
+const toggleTodo = currentListElement => {
     const currentObject = getObjectFromDOMElement(currentListElement)
+    const [checkBox,] = currentListElement.getElementsByClassName('checkBox')
```
aa

jpudysz commented on index.js:
```diff
@@ -1,115 +1,143 @@
+const toggleTodo = currentListElement => {
     const currentObject = getObjectFromDOMElement(currentListElement)
+    const [checkBox,] = currentListElement.getElementsByClassName('checkBox')
+    const [toDoText,] = currentListElement.getElementsByClassName('textField')
```
aa

jpudysz commented on index.js:
```diff
@@ -1,115 +1,143 @@
+        toDoText .style.setProperty('text-decoration', '');
+        currentObject.toggle()
         printItems()
+        return
```
enter before `return`

jpudysz commented on index.js:
```diff
@@ -1,115 +1,143 @@
 
-const printItems = () => document.getElementById('debugger').innerText=JSON.stringify([...toDoObjectsArray.entries()])
+const hideElement = (name, div) => {
+    const [elementToHide,] = div.getElementsByClassName(name)
```
aa

jpudysz commented on index.js:
```diff
@@ -1,115 +1,143 @@
+    const [elementToHide,] = div.getElementsByClassName(name)
 
-const hideElement = (name, div) => div.getElementsByClassName(name)[0].setAttribute('hidden','true')
+    elementToHide.setAttribute('hidden',true)
```
aa

jpudysz commented on index.js:
```diff
@@ -1,115 +1,143 @@
 
-const showElement = (name, div) => div.getElementsByClassName(name)[0].removeAttribute('hidden')
+const showElement = (name, div) => {
+    const [elementToShow,] = div.getElementsByClassName(name)
```
aa

jpudysz commented on index.js:
```diff
@@ -1,115 +1,143 @@
+
+const hideButtons = currentListElement => currentListElement
+    .querySelectorAll('button')
+    .forEach(button => button.setAttribute('hidden',true))
```
aa

jpudysz commented on index.js:
```diff
@@ -1,115 +1,143 @@
+    .querySelectorAll('button')
+    .forEach(button => button.setAttribute('hidden',true))
+
+const hideOptions = div => div.setAttribute('hidden',true)
```
aa

jpudysz commented on index.js:
```diff
@@ -118,32 +146,32 @@ const newListElement = (text,id) => {
 
     const inputField = document.createElement('input')
-    inputField.setAttribute('hidden','true')
+    inputField.setAttribute('hidden',true)
```
aa

jpudysz commented on index.js:
```diff
@@ -118,32 +146,32 @@ const newListElement = (text,id) => {
 
     const saveEditButton = document.createElement('button')
-    saveEditButton.setAttribute('hidden','true')
+    saveEditButton.setAttribute('hidden',true)
```
aa

jpudysz commented on index.js:
```diff
@@ -152,22 +180,19 @@ const newListElement = (text,id) => {
     div.appendChild(editButton)
     div.appendChild(deleteButton)
-    div.setAttribute('hidden','true')
+    div.setAttribute('hidden',true)
```
aa

mmarganski committed:
refactored second time;

mmarganski committed:
changed to object-based todo list

jpudysz commented on index.js:
```diff
@@ -1,197 +1,240 @@
-        this.isDone = false
-        this.text = input
+    constructor() {
+        this.todos = new Todos(() => this.printItems())
```
proxy, remember about this:
```
a(b) is equal to a(() => b())
```

jpudysz commented on index.js:
```diff
@@ -1,197 +1,240 @@
-        this.text = input
+    constructor() {
+        this.todos = new Todos(() => this.printItems())
+        document.getElementById('mainSubmitButton').setAttribute('onclick', 'myApp.createNewTodo()')
```
remember that you're in JS wolrd. You can use reference to the function rather than string eg.
```
this.clearAllTasks
```

jpudysz commented on index.js:
```diff
@@ -1,197 +1,240 @@
-    if(!checkBox.checked) {
+            this.showElement('editButton', currentListElement)
+            this.showElement('deleteButton', currentListElement)
+            toDoText .style.setProperty('text-decoration', '')
```
remove space

jpudysz commented on index.js:
```diff
@@ -1,197 +1,240 @@
-    printItems()
-}
+    createNewListElement = (text, id) => {
+
```
remove enter

jpudysz commented on index.js:
```diff
@@ -1,197 +1,240 @@
+        const strings = []
 
-    elementToHide.setAttribute('hidden', true)
+        this.toDoObjectsMap.forEach( todo => strings.push(todo.toString()))
```
remove space

mmarganski committed:
furhter refactor

mmarganski committed:
converted to jQuery and split into multiple files

mmarganski committed:
converted to jQuery and split into multiple files

jpudysz commented on src/app.js:
```diff
@@ -0,0 +1,37 @@
+import Todos from "./todos.js";
```
please use single quote across all files

jpudysz replied:
add default export from `todos.js`

jpudysz commented on src/app.js:
```diff
@@ -0,0 +1,37 @@
+import Todos from "./todos.js";
+
+class App {
+
```
remove enter

jpudysz commented on src/app.js:
```diff
@@ -0,0 +1,37 @@
+    }
+}
+
+export default App
```
normal export is fine

jpudysz commented on src/index.js:
```diff
@@ -0,0 +1,6 @@
+import App from './app.js'
+import Todos from './todos.js'
+import Todo from "./todo.js"
```
be consistent, use single quote

jpudysz commented on src/index.js:
```diff
@@ -0,0 +1,6 @@
+import Todos from './todos.js'
+import Todo from "./todo.js"
+
+const myApp = new App()
```
`new App()` is more than enough as you don't use myApp later

jpudysz commented on src/todo.js:
```diff
@@ -0,0 +1,32 @@
+class Todo {
+
```
aa

jpudysz commented on src/todo.js:
```diff
@@ -0,0 +1,32 @@
+    }
+}
+
+export default Todo
```
aa

jpudysz commented on src/todo.js:
```diff
@@ -0,0 +1,32 @@
+    constructor(input) {
+        this.id = this.#createID()
+        this.#isDone = false
+        this.#text = typeof input === "string"
```
aa

jpudysz commented on src/todos.js:
```diff
@@ -0,0 +1,127 @@
+import Todo from "./todo.js";
```
aa

jpudysz commented on src/todos.js:
```diff
@@ -0,0 +1,127 @@
+import Todo from "./todo.js";
+
+class Todos {
+
```
aa

jpudysz commented on src/todos.js:
```diff
@@ -0,0 +1,127 @@
+    }
+
+    deleteTodo = id => {
+
```
remove enter

jpudysz commented on src/todos.js:
```diff
@@ -0,0 +1,127 @@
+    }
+}
+
+export default Todos
```
aa

jpudysz commented:
Please add at least tests for `Todo.js`

mmarganski committed:
added test files

mmarganski committed:
add test files

mmarganski committed:
resolved conflicts

jpudysz commented on index.test.js:
```diff
@@ -0,0 +1,58 @@
+const {Todo} = require('./src/todo')
```
spaces between {}

jpudysz replied:
index.text.js should be inside src next to target file

jpudysz commented on index.test.js:
```diff
@@ -0,0 +1,58 @@
+        const testTodo1 = new Todo('lorem ipsum')
+        expect(testTodo1.text).toBe(undefined)
+    } )
+})
```
nice tests

jpudysz commented on src/app.js:
```diff
@@ -34,4 +33,4 @@ class App {
 }
 
-export default App
+export {App}
```
you can write `export class App` instead `export {App}`

jpudysz commented on src/index.js:
```diff
@@ -1,6 +1,6 @@
-import App from './app.js'
-import Todos from './todos.js'
-import Todo from "./todo.js"
+import {App} from './app.js'
```
spaces between brackets

jpudysz commented on src/todo.js:
```diff
@@ -25,8 +24,8 @@ class Todo {
 }
 
-export default Todo
+export {Todo}
```
aa

jpudysz commented on src/todos.js:
```diff
@@ -1,7 +1,6 @@
-import Todo from "./todo.js";
+import {Todo} from './todo.js';
```
aa

jpudysz commented on src/todos.js:
```diff
@@ -124,4 +122,4 @@ class Todos {
 }
 
-export default Todos
+export {Todos}
```
aa

mmarganski committed:
refactored imports, exports

jpudysz commented:
You can merge it as I don't have a write access
